components/com_joomgallery/models/image.php

```diff
@@ -262,7 +262,70 @@ public function includeWatermark($file, $src_img = null, $cropwidth = 0, $crophe
     {
       $info_img = array(0 => imagesx($src_img), 1 => imagesy($src_img));
     }
-    $info_wat = getimagesize($watermark);
+   $info_wat      = getimagesize($watermark);
+    $widthwm       = $info_wat[0];
+    $heightwm      = $info_wat[1];
+    $watermarksize = $this->_config->get('jg_watermarksize');
+    $watermarkzoom = $this->_config->get('jg_watermarkzoom');
+
+    switch($watermarkzoom) 
+    {
+      // no resize
+      case 1:
+        $newheight_watermark    = $heightwm;
+        $newwidth_watermark     = $widthwm;
+        break;
+      // by height
+      case 2:
+        $newheight_watermark    = $info_img[1] * $watermarksize / 100;
+        $newwidth_watermark     = $newheight_watermark * $widthwm / $heightwm;
+        if($newwidth_watermark  > $info_img[0])
+        {
+          $newwidth_watermark   = $info_img[0];  
+        }
+        break;
+      // by width
+      case 3:
+        $newwidth_watermark     = $info_img[0] * $watermarksize / 100;
+        $newheight_watermark    = $newwidth_watermark * $heightwm / $widthwm;
+        if($newheight_watermark > $info_img[1])
+        {
+          $newheight_watermark  = $info_img[1];
+        }
+        break;
+      default:
+        $newheight_watermark    = $heightwm;
+        $newwidth_watermark     = $widthwm;
+        break;
+    }
+
+    switch($info_wat[2])
+    {
+      case 1:
+        $watermark  = imagecreatefromgif($watermark);
+        $mime_wat   = 'image/gif';
+        break;
+      case 2:
+        $watermark  = imagecreatefromjpeg($watermark);
+        $mime_wat   = 'image/jpeg';
+        break;
+      case 3:
+        $watermark  = imagecreatefrompng($watermark);
+        $mime_wat   = 'image/png';
+        break;
+      default:
+        $this->setError(JText::sprintf('COM_JOOMGALLERY_COMMON_MSG_MIME_NOT_ALLOWED', $info_wat[2]));
+
+        return false;
+    }
+          
+    $newwatermark = ImageCreateTrueColor($newwidth_watermark, $newheight_watermark);
```

Comment:
Die Berechnungen in den Zeilen 322-328 sind im Fall keiner Größenänderung doch eigentlich gar nicht notwendig, oder?